swatch-contracts/ct/java/tests/ContractsComponentTest.java

```diff
@@ -0,0 +1,201 @@
+        "Contract retrieval call should succeed", getContractsResponse.statusCode(), is(200));
+
+    getContractsResponse
+        .then()
+        .body("size()", equalTo(1))
+        .body("[0].org_id", equalTo(orgId))
+        .body("[0].subscription_number", equalTo(subscriptionNumber))
+        .body("[0].billing_account_id", equalTo(awsAccountId))
+        .body("[0].sku", equalTo(testSku))
+        .body("[0].metrics", notNullValue())
+        .body("[0].metrics.size()", greaterThan(0));
+  }
+
+  /** Verify pure pay-as-you-go ROSA contract is created when all dimensions are incorrect. */
+  @Test
+  @Tag("contract")
+  void shouldCreatePurePaygRosaContract_whenAllDimensionsAreIncorrect() {
+    String orgId = "org234";
+    String subscriptionId = "sub14968327";
+    String subscriptionNumber = "24968327";
+    String awsCustomerId = "CCwCpt6sqkC";
+    String awsAccountId = "268056954830";
+    String productCode = "2n58d3s3qpvk22dgew2gal7w3";
+    String metricValue = "10";
+    String sellerAccountId = "223456789";
+    String sourcePartner = "aws_marketplace";
+    String billingProvider = "aws";
+    String productTag = "rosa";
+    String testSku = "TEST1MW02393";
+    String testMetricName = "premium_support"; // Invalid metric for ROSA
+
+    OfferingTestData offeringData = buildRosaOffering(testSku);
+    ContractTestData contractData =
+        buildRosaContract(
+            testSku,
+            testMetricName,
+            orgId,
+            subscriptionId,
+            subscriptionNumber,
+            awsCustomerId,
+            awsAccountId,
+            productCode,
+            metricValue,
+            sellerAccountId,
+            sourcePartner);
+
+    offeringWiremock.stubOfferingData(offeringData);
+    contractsWiremock.stubPartnerEntitlement(contractData);
+
+    // Sync offering needed for contract to persist with the SKU
+    Response syncOfferingResponse = OfferingTestHelper.syncOffering(service, testSku);
+    assertThat("Sync offering call should succeed", syncOfferingResponse.statusCode(), is(200));
+
+    Response createContractResponse = ContractsTestHelper.createContract(service, contractData);
+    assertThat(
+        "Prepaid contract creation should succeed", createContractResponse.statusCode(), is(200));
+    service.logs().assertContains("Creating contract");
+
+    // Retrieve and verify contract
+    Response getContractsResponse =
+        ContractsTestHelper.getContracts(service, orgId, billingProvider, awsAccountId, productTag);
+    assertThat(
+        "Contract retrieval call should succeed", getContractsResponse.statusCode(), is(200));
+
+    getContractsResponse
+        .then()
+        .body("size()", equalTo(1))
+        .body("[0].org_id", equalTo(orgId))
+        .body("[0].subscription_number", equalTo(subscriptionNumber))
+        .body("[0].billing_account_id", equalTo(awsAccountId))
+        .body("[0].sku", equalTo(testSku))
+        .body("[0].metrics.size()", equalTo(0));
```

Comment:
Having metrics size be zero is what is indicating that this is pure paygo because there are no prepaid metric amounts. This might be good to call out in a comment so that someone that comes back later understands why it is important.